server/src/main/java/io/crate/expression/scalar/cast/ImplicitCastFunction.java

```diff
@@ -47,65 +42,53 @@ public class ImplicitCastFunction extends Scalar<Object, Object> {
 
     public static final String NAME = "_cast";
     public static final Signature SIGNATURE = Signature.builder(NAME, FunctionType.SCALAR)
+        .argumentTypes(TypeSignature.parse("E"), TypeSignature.parse("V"))
+        .returnType(TypeSignature.parse("V"))
+        .features(Feature.DETERMINISTIC)
+        .typeVariableConstraints(typeVariable("E"))
```

Comment:
I wonder if this is needed at all as afaik the `typeVariableConstraints` is only meant to ensure that multiple usages inside the argument list of the same placeholder/variable must fit to the same actual type. (or to ensure that all will be implicit cast/converted to the same type)

Reply:
Already started working on removing the second arg https://github.com/crate/crate/commit/0d321daf0fd4f7adbd91bbcab6bdba1cc3844a62 - but would keep that as dedicated follow up, because it needs bwc signatures for try and explicit cast too.